jOOQ-kotlin/src/main/kotlin/org/jooq/kotlin/Extensions.kt

```diff
@@ -220,3 +233,59 @@ inline fun <reified T: Any> Select<Record1<T>>.notIn(vararg values: T): Conditio
 
 @Support
 inline fun <reified T: Any> Select<Record1<T>>.notIn(vararg values: Field<*>): Condition = field(this).notIn(values.asList())
+
+// ----------------------------------------------------------------------------
+// Extensions to make mapping into objects more idiomatic
+// ----------------------------------------------------------------------------
+
+@Support
+inline fun <reified T: Any> Record.into() = this.into(T::class.java)
+
+@Support
+inline fun <reified T: Any> Result<*>.into() = this.into(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchInto() = this.fetchInto(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchAnyInto() = this.fetchAnyInto(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchOneInto() = this.fetchOneInto(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchOptionalInto() = this.fetchOptionalInto(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchSingleInto() = this.fetchSingleInto(T::class.java)
+
+@Support
+inline fun <reified T: Any> Fetchable<*>.fetchStreamInto() = this.fetchStreamInto(T::class.java)
+
+// ----------------------------------------------------------------------------
+// Extensions to integrate async methods with coroutines
+// ----------------------------------------------------------------------------
+
+@Support
+suspend fun <R: Record> Fetchable<R>.fetchAwait(executor: Executor): Result<R> = suspendCoroutine {
+        this.fetchAsync(executor)
+                .handle { result, error -> if (error != null) it.resumeWithException(error) else it.resume(result) }
+}
+
+@Support
+suspend fun <R: Record> Fetchable<R>.fetchAwait(): Result<R> = suspendCoroutine {
+        this.fetchAsync()
+                .handle { result, error -> if (error != null) it.resumeWithException(error) else it.resume(result) }
+}
+
+@Support
+suspend fun <R: Record> Query.executeAwait(executor: Executor): Int = suspendCoroutine {
+        this.executeAsync(executor)
+                .handle { result, error -> if (error != null) it.resumeWithException(error) else it.resume(result) }
+}
+
+@Support
+suspend fun Query.executeAwait(): Int = suspendCoroutine {
+        this.executeAsync()
+                .handle { result, error -> if (error != null) it.resumeWithException(error) else it.resume(result) }
+}
```

Comment:
These kinds of things are being discussed here: https://github.com/jOOQ/jOOQ/issues/9335

I don't feel at ease with the status quo of having:

- Classic blocking API (which will "soon" be supercharged by Loom)
- Asynchronous API via JDK `CompletionStage`
- Reactive API via reactive-streams or JDK `Flow`
- Kotlin specific coroutine API

All of jOOQ's API needs to be duplicated many times, with a lot cognitive overhead for users and maintenance overhead on our side.

I'm still betting on Loom, hoping to sit out all of the above fads, which are *much more* thoroughly addressed by Loom, without the need to change a single thing to jOOQ's API (or our internal JDBC usage).

Reply:
Thanks again for the link. I'll add my two cents there.

Reply:
Currently the Select is a `Publisher`, it is easy to convert to Kotlin Coroutines via Reactor kotlin extensions.

Reply:
Can you show an example, @hantsy ?